src/App.tsx

```diff
@@ -1,14 +1,66 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [status, setStatus] = useState<string>('all');
```

Comment:
1) If you pass the type as (string, number, boolean) - the compiler understands what type it is and you don't need to type it.
2) u created a generic for this. Use it in ur state
```suggestion
  const [query, setQuery] = useState('');
  const [status, setStatus] = useState<Filters>(Filters.ALL);
```